src/Simplify.elm

```diff
@@ -8788,6 +8791,36 @@ setFoldrChecks =
         )
 
 
+setFoldChecks : CallCheckInfo -> Maybe (Error {})
+setFoldChecks checkInfo =
+    emptiableFoldChecks setCollection checkInfo
+        |> onNothing
+            (\() ->
+                if checkInfo.expectNaN then
```

Comment:
```elm
a = Set.fromList [NaN,NaN,NaN]

Set.foldl Set.insert Set.empty a
-->
Set.fromList [NaN,NaN,NaN]
```

These are the same, though it now fails a `a == <result>` check. I think all cases where we can make `==` return `True` instead is better, but maybe that is just a biased opinion. I'm missing use-cases where having `==` return `False` in this sort of case is useful :man_shrugging: 

Maybe let's keep it the way you wrote it, that's "safer", but yeah... :man_shrugging: